File: is_int/still_working/go/is_int.go
package main

import (
	"fmt"
)

func isInt(s string) bool {
	var r = []rune(s)
	l := len(r)
	i := 0

	goto S0
S0:
	if i >= l {
		goto S4
	} else if r[i] == '+' || r[i] == '-' {
		i++
		goto S1
	} else if '0' <= r[i] && r[i] <= '9' {
		i++
		goto S2
	} else {
		goto S4
	}
S1:
S2:
S3:
	return true
S4:
	return false
}

func main() {
	tests := []struct {
		s string
		b bool
	}{
		{"", false},
		{" ", false},
		{"+", false},
		{"-", false},
		{"++", false},
		{"--", false},
		{"+0+", false},
		{"-0-", false},
		{"+0123456789", true},
		{"-0123456789", true},
		{"0123456789", true},
	}
	for _, t := range tests {
		fmt.Printf("%s\t'%s'\n", map[bool]string{true: "OK", false: "NG"}[t.b == isInt(t.s)], t.s)
	}
}

File: is_int/still_working/julia/is_int.jl
function is_int(s)
	i = firstindex(s)
	l = lastindex(s)
	@goto S0

	@label S0
	if i > l 
		@goto S4
	elseif s[i] == '+' || s[i] == '-'
		i = nextind(s, i)
		@goto S1
	elseif '0' <= s[i] && s[i] <= '9'
		i = nextind(s, i)
		@goto S2
	else
		@goto S4
	end

	@label S1

	@label S2

	@label S3
	return true

	@label S4
	return false
end

tests = [
	 ("", false)
	 (" ", false)
	 ("+", false)
	 ("-", false)
	 ("++", false)
	 ("--", false)
	 ("+0+", false)
	 ("-0-", false)
	 ("+0123456789", true)
	 ("-0123456789", true)
	 ("0123456789", true)
	 ]

for t in tests
    println("$(t[2] == is_int(t[1]) ? "OK" : "NG")\t'$(t[1])'")
end
